src/main/java/com/aws/greengrass/device/ClientDevicesAuthService.java

```diff
@@ -99,9 +96,27 @@ public ClientDevicesAuthService(Topics topics, GroupManager groupManager, Certif
     protected void install() throws InterruptedException {
         super.install();
         //handleConfiguration
-        this.deviceGroupsTopics = this.config.lookupTopics(CONFIGURATION_CONFIG_KEY, DEVICE_GROUPS_TOPICS);
-        this.deviceGroupsTopics.subscribe(this::handleConfigurationChange);
-        this.config.lookup(CONFIGURATION_CONFIG_KEY, CA_TYPE_TOPIC).subscribe(this::updateCAType);
+        this.config.lookupTopics(CONFIGURATION_CONFIG_KEY).subscribe((whatHappened, node) -> {
+            if (whatHappened == WhatHappened.timestampUpdated || whatHappened == WhatHappened.interiorAdded) {
+                return;
+            }
+            logger.atDebug().kv("why", whatHappened).kv("node", node).log();
+            Topics deviceGroupTopics = this.config.lookupTopics(CONFIGURATION_CONFIG_KEY, DEVICE_GROUPS_TOPICS);
+            Topic caTypeTopic = this.config.lookup(CONFIGURATION_CONFIG_KEY, CA_TYPE_TOPIC);
+
+            if (whatHappened == WhatHappened.initialized) {
+                handleConfigurationChange(whatHappened, deviceGroupTopics);
+                updateCAType(whatHappened, caTypeTopic);
+            } else if (DEVICE_GROUPS_TOPICS.equals(node.getName())) {
+                handleConfigurationChange(whatHappened, deviceGroupTopics);
+            } else if (CA_TYPE_TOPIC.equals(node.getName())) {
+                if (caTypeTopic.toPOJO() == null) {
+                    return;
+                }
```

Comment:
When can this case happen?
Can this happen for deviceGroupTopics too? I didnt see that check

Reply:
It happens after initialization when the structure of the config is being generated. A null child is added initially.